src/test/scala/chiselTests/MultiClockSpec.scala

```diff
@@ -142,6 +142,18 @@ class MultiClockSpec extends ChiselFlatSpec {
       assert(withReset(this.reset) { 5 } == 5)
     })
   }
+  it should "support literal Bools" in {
```

Comment:
This test seems kind of random, I assume this fix is a side effect of the above change?

Reply:
#763 was the motivating issue so I added a test for it :) In looking into that issue, I realized that literal Bools not working in `withReset` was a side effect of the weirdness that this PR fixes